### PR TITLE
Support backup and restore with ObjectBucketName set in environment variables

### PR DESCRIPTION
**Pull request by yamatcha**
Currently, there is no way to set the `ObjectBucketName` in ConfigMap or Secrets. This can be accomplished by using envFrom without adding new functionality.
So we have added docs and some test cases in this PR.

Signed-off-by: yamatcha <soju-yamashita@cybozu.co.jp>

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -457,6 +457,27 @@ spec:
 ...
 ```
 
+>**Note:** If you want to specify the ObjectBucket name in configMap or secret, you can use envFrom and specify environments variable in `ObjectBucketName` like below.
```
Comment (ymmt2005) [inline docs/usage.md:460]: Do you mean the below?
```suggestion
>**Note:** If you want to specify the ObjectBucket name in a ConfigMap or Secret, you can use `envFrom` and specify the environment variable name in `jobConfig.bucketConfig.bucketName` as follows.
```

File: docs/usage.md [hunk 2]
```diff
@@ -457,6 +457,27 @@ spec:
...
+spec:
+  jobConfig:
+    bucketConfig:
+      bucketName: "$(BUCKET_NAME)"
+      endpointURL: http://minio.default.svc:9000
+      usePathStyle: true
+    envFrom:
+    - configMapRef:
+        name: bucket-name
+...
```
Comment (ymmt2005) [inline docs/usage.md:478]: It'd be more helpful if we include an example ConfigMap.
```yaml
...
---
apiVersion: v1
kind: ConfigMap
data:
  BUCKET_NAME: foobar

File: docs/usage.md (hunk 1 above)
Comment (yamatcha) [reply docs/usage.md:460]: @ymmt2005 
Yes, thank you for pointing that out.

**Review by kmdkuk — APPROVED** [review]

**Pull request merged**